src/result_processor.c

```diff
@@ -891,6 +898,15 @@ static int rpSafeLoaderNext_Accumulate(ResultProcessor *rp, SearchResult *res) {
   // Done loading. Unlock Redis
   RedisModule_ThreadSafeContextUnlock(sctx->redisCtx);
 
+  if (isQueryProfile) {
+    clock_gettime(CLOCK_MONOTONIC, &rpEndTime);
+  } else {
+    rpEndTime.tv_nsec = 0;
+  }
+  double GILDuration =  (rpEndTime.tv_nsec - rpStartTime.tv_nsec)/1000000.0;
+  rp->parent->GILTime += GILDuration;
+  rp->GILTime += GILDuration;
```

Comment:
consider doing these operations only if `isQueryProfile`, and not always with 0 values